pkg/app/piped/controller/controller.go

```diff
@@ -643,6 +671,19 @@ func (c *controller) getMostRecentlySuccessfulDeployment(ctx context.Context, ap
 	return nil, err
 }
 
+func (c *controller) shouldStartPlanningDeployment(ctx context.Context, d *model.Deployment) (bool, error) {
+	if !d.IsInChainDeployment() {
+		return true, nil
+	}
+	resp, err := c.apiClient.InChainDeploymentPlannable(ctx, &pipedservice.InChainDeploymentPlannableRequest{
+		Deployment: d,
+	})
+	if err != nil {
+		return false, err
+	}
+	return resp.Plannable, nil
+}
+
```

Comment:
How about updating this to be an independent function instead of Controller's one?

Reply:
Not sure what is better that way? Since we need the `c.apiClient` to make RPC call.

Reply:
Ok, let's keep it as it is. I don't have a strong opinion on this.